cli/command/plugin/upgrade.go

```
@@ -64,10 +64,7 @@ func runUpgrade(ctx context.Context, dockerCli command.Cli, opts pluginOptions)
 	fmt.Fprintf(dockerCli.Out(), "Upgrading plugin %s from %s to %s\n", p.Name, reference.FamiliarString(old), reference.FamiliarString(remote))
 	if !opts.skipRemoteCheck && remote.String() != old.String() {
 		if r, err := command.PromptForConfirmation(ctx, dockerCli.In(), dockerCli.Out(), "Plugin images do not match, are you sure?"); !r || err != nil {
-			if err != nil {
-				return errors.Wrap(err, "canceling upgrade request")
-			}
-			return errors.New("canceling upgrade request")
+			return err
```

todo: ensure an error is returned, if `!r` it should return a cancelled error.